add tab extension to dev console (#12109)
openshift/console: frontend/packages/console-dynamic-plugin-sdk/docs/console-extensions.md

---

## `console.tab`

### Summary 

> Contributes a tab into an HorizontalNav component matching the sectionId.

Reviewer: ```suggestion
Adds a tab to an `HorizontalNav` matching the `contextId`.
```

Reviewer: A code example would be good here.

Reviewer: We should discuss how ordering works.

Reviewer 2: `a HorizontalNav`

Author: I changed and fixed the text, unfortunately when adding TSDoc `@example` the docs are not generated correctly. I checked the generation but it seems to be a task big enough for a different PR.

### Properties

| Name | Value Type | Optional | Description |
| ---- | ---------- | -------- | ----------- |

> | `contextId` | `string` | no | ContextId assigned to the horizontal nav in which the tab will be injected |

Reviewer: ```suggestion
| `contextId` | `string` | no | Context ID assigned to the horizontal nav in which the tab will be injected |
```

Reviewer: Should we list the known context IDs?

Reviewer 2: Inconsistencies in docs with `HorizontalNav` and `horizontal nav`

Author: Added a list of possible values. Decided for horizontal nav in the docs.

> | `page` | `{ name: string; href: string; }` | no | The page to be show in horizontal tab. It takes tab name as name and href of the tab |

Reviewer: We should clarify what `name` is. Is this the label of the tab or some internal name?

Reviewer: When we use `href` here, what do we mean? Is it just the path appended to the existing URL? (If so, I'm not sure `href` is the right property name, but we probably use that elsewhere.)

Author: I added more descriptive docs. The `href` is the path appended to the existing URL, used elsewhere.
